packages/rest/src/controllers/didcomm/connections/ConnectionsControllerTypes.ts

```diff
@@ -0,0 +1,58 @@
+import type { Did } from '../../did/DidsControllerTypes'
+import type { CredoBaseRecord, ThreadId } from '../../types'
+import type {
+  ConnectionType,
+  ConnectionRecord as CredoConnectionRecord,
+  DidExchangeRole,
+  DidExchangeState,
+  HandshakeProtocol,
+} from '@credo-ts/core'
+
+export interface DidCommConnectionsRecord extends CredoBaseRecord {
+  did?: Did
+  theirDid?: Did
+  theirLabel?: string
+  state: DidExchangeState
+  role: DidExchangeRole
+  alias?: string
+  autoAcceptConnection?: boolean
+  threadId?: ThreadId
+  imageUrl?: string
+  mediatorId?: string
+  errorMessage?: string
+  protocol?: HandshakeProtocol
+  outOfBandId?: string
+  invitationDid?: Did
+  connectionTypes?: Array<ConnectionType | string>
+  previousDids?: Array<Did>
+  previousTheirDids?: Array<Did>
+}
+
+export function connectionRecordToApiModel(record: CredoConnectionRecord): DidCommConnectionsRecord {
+  return {
```

Comment:
We can't use the record.toJSON() for this?

Reply:
Because it will include tags / metadata and some other fields and it messed with the types. So I'd rather have a simple facade that makes sure the Swagger Type makes sense and isn't bloated with internal storage propertiees.

So mainly preference on my side to be explicity because all the tsoa magic and other things add a lot of weird behaviour